### PR TITLE
Todo

### PR DESCRIPTION
**Pull request by RakshithBhat03**
# Todolist

## Features
- Add new todo.
- Check/uncheck todo.
- Delete todo.
- Data persistence using local storage.

## Preview: https://deploy-preview-7--clutch-views.netlify.app/

### CODE REVIEW

File: src/reducers/todoReducer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { v4 as uuid } from "uuid";
+export const todoReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TODO":
+      const newTodo = [
+        ...state,
+        { id: uuid(), isChecked: false, content: action.payload },
```
Comment (junaidshaikh-js) [inline src/reducers/todoReducer.js]: `action.payload` can be destructured in the parameter.

File: src/reducers/todoReducer.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        ? localStorage.setItem("todo", JSON.stringify(removedTodo))
+        : localStorage.removeItem("todo");
+      return removedTodo;
+    case "TOGGLE_CHECKED":
+      const checkedTodo = state.map((todo) =>
+        todo.id === action.payload
+          ? { ...todo, isChecked: !todo.isChecked }
+          : todo
+      );
+      localStorage.setItem("todo", JSON.stringify(checkedTodo));
```
Comment (junaidshaikh-js) [inline src/reducers/todoReducer.js]: The logic here can be extracted into a function to reduce the complexity of the code written to make it more readable.

**Review by junaidshaikh-js — COMMENTED** [review]
Good work done. 

Suggested some changes, everything else looks good.

**Comment by aryan-13** [comment]
Changes suggested by @junaidshaikh-js can be implemented, Else everything looks fine.
Great work!

**Review by aryan-13 — APPROVED** [review]
Changes suggested by @junaidshaikh-js can be implemented, Else everything looks fine.
Great work!

**Pull request merged**